server/auth/types/basic/basic_auth.ts

```diff
@@ -104,74 +95,52 @@ export class BasicAuthentication implements IAuthenticationType {
-    }
-
-    let cookie: SecuritySessionCookie | null;
-    try {
-      cookie = await this.sessionStorageFactory.asScoped(request).get();
-      // TODO: need to do auth for each all?
-      if (!cookie || !cookie.credentials) {
-        if (request.url.pathname === '/' || request.url.pathname?.startsWith('/app')) {
-          // requesting access to an application page, redirect to login
-          const nextUrlParam = this.composeNextUrlQeuryParam(request);
-          const redirectLocation = `${this.coreSetup.http.basePath.serverBasePath}/app/login?${nextUrlParam}`;
-          return response.redirected({
-            headers: {
-              location: `${redirectLocation}`,
-            },
-          });
-        }
-        return response.unauthorized();
-      }
-
-      const headers = {};
-
-      // set cookie to extend ttl
-      cookie.expiryTime = Date.now() + this.config.cookie.ttl;
-      this.sessionStorageFactory.asScoped(request).set(cookie);
-
-      // pass credentials to request to Elasticsearch
-      Object.assign(headers, { authorization: cookie.credentials?.authHeaderValue });
-
-      // add tenant to Elasticsearch request headers
-      if (this.config.multitenancy?.enabled && isMultitenantPath(request)) {
-        // TODO: consider move tenant resolution to postAuthHandler?
-        const authInfo = await this.securityClient.authinfo(request, headers);
-        const selectedTenant = resolveTenant(
-          request,
-          authInfo.user_name,
-          authInfo.tenants,
-          this.config,
-          cookie
-        );
-        Object.assign(headers, { securitytenant: selectedTenant });
-
-        if (selectedTenant !== cookie.tenant) {
-          cookie.tenant = selectedTenant;
-          this.sessionStorageFactory.asScoped(request).set(cookie);
-        }
-      }
-
-      return toolkit.authenticated({
-        // state: credentials,
-        requestHeaders: headers,
-      });
-    } catch (error) {
-      this.logger.error(`Failed authentication: ${error}`);
-      return toolkit.notHandled();
-      // TODO: redirect using response?
-    }
-  };
+  // override functions inherited from AuthenticationType
+  requestIncludesAuthInfo(request: KibanaRequest<unknown, unknown, unknown, any>): boolean {
+    return request.headers[BasicAuthentication.AUTH_HEADER_NAME] ? true : false;
+  }
+
+  getAdditionalAuthHeader(request: KibanaRequest<unknown, unknown, unknown, any>) {
+    return {};
+  }
+
+  getCookie(request: KibanaRequest, authInfo: any): SecuritySessionCookie {
+    return {
+      username: authInfo.user_name,
+      credentials: {
+        authHeaderValue: request.headers[BasicAuthentication.AUTH_HEADER_NAME],
+      },
+      authType: this.type,
+      expiryTime: Date.now() + this.config.cookie.ttl,
+    };
+  }
+
+  isValidCookie(cookie: SecuritySessionCookie): boolean {
+    return (
+      cookie.authType === this.type &&
+      cookie.username &&
+      cookie.expiryTime &&
```

Comment:
For validation purpose, shall we check it is >= Date.now()

Reply:
that has been validated by cookie validator at https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/blob/dev/server/session/security_cookie.ts#L70